apps/docs/src/content/learn/advanced/webgpu.mdx

```diff
@@ -0,0 +1,192 @@
+`<Canvas>`'s `createRenderer` prop.
+
+```svelte title="App.svelte" {4}+ {8-14}+
+<script>
+  import Scene from './Scene.svelte'
+  import { Canvas } from '@threlte/core'
+  import * as THREE from 'three/webgpu'
+</script>
+
+<Canvas
+  createRenderer={(canvas) => {
+    return new THREE.WebGPURenderer({
+      canvas,
+      antialias: true,
+      forceWebGL: false
+    })
+  }}
+>
+  <Scene />
+</Canvas>
+```
+
+<Tip type="note">
+  WebGPU is still an experimental browser api and at the time of writing has 73% global
+  [support](https://caniuse.com/?search=webgpu). For this reason, Three.js's webgpu renderer
+  fallbacks to webgl when webgpu is not available.
+</Tip>
+<Tip type="tip">
+  This same approach can be used to swap out the default renderer for any other custom renderer.
+</Tip>
+
+<Example path="renderers/WebGPU" />
+
+<small>
+  Adapted from [this Three.js
+  example](https://threejs.org/examples/?q=webgpu#webgpu_performance_renderbundle).
+</small>
+
+The WebGPU renderer doesn't immediately render. If the renderer you provide needs to delay renderering, you
+can defer rendering by initially setting the renderMode to `manual` like so:
+
+```svelte title="App.svelte"
+<script>
+  import { Canvas, T } from '@threlte/core@next'
+  import * as THREE from 'three/webgpu'
+  let renderMode = $state('manual')
+</script>
+
+<Canvas
+  {renderMode}
+  createRenderer={(canvas) => {
+    const renderer = new THREE.WebGPURenderer({
+      canvas,
+      antialias: true,
+      forceWebGL: false
+    })
+    renderer.init().then(() => {
+      renderMode = 'on-demand'
+    })
+    return renderer
+  }}
+>
+  <Scene />
+</Canvas>
+```
+
+### Vite
+
+WebGPU uses top-level async to determine WebGPU compatibility. Vite
+will often throw an error when it detects this.
+
+To circumvent this issue, the following can be added to your Vite config.
+
+```js
+// vite.config.js
+optimizeDeps: {
+  esbuildOptions: {
+    target: 'esnext'
+  }
+},
+build: {
+  target: 'esnext'
+}
+```
+
+Alternatively,
+[`vite-plugin-top-level-await`](https://github.com/Menci/vite-plugin-top-level-await)
+can be used, although less success has been reported with this method.
+
+## TSL
+
+A question that comes up often in Three.js development is "How do I extend Three.js's materials?".
+External libraries such as [three-custom-shader-material](https://www.npmjs.com/package/three-custom-shader-material)
+use a find and replace solution to get this job done. Three.js has identified
+that it's not an ideal solution and recommends using the [Three.js Shading Language](https://github.com/mrdoob/three.js/wiki/Three.js-Shading-Language) or TSL for short.
+
+The example below is an adaptation of [this](https://threejs.org/examples/?q=tsl#webgpu_tsl_angular_slicing)
+Three.js example. There are many more [TSL examples](https://threejs.org/examples/?q=tsl)
+within Three.js that you can use or adapt for your project.
+
+<Example path="shaders/slice/main" />
+
+### Extend the catalogue
```

Comment:
Even though this is convenient it still has big implications for bundle size. I'd recommend warning about this caveat and [also providing the alternative method that verekia came up with](https://github.com/verekia/three-gpu-ecosystem-tests/blob/main/vite-ts-threlte8/src/App.svelte) (which I'd use - and probably always recommend for stuff that's going into production considering how much unused code is imported with this method)

Reply:
Hmm I think I'd like to talk about both. Maybe start with the easy Laissez-faire option of going wholesale and then talking about caveats in a sentence or two afterwards. It's a good point and grisch said verekia had tested it out too.

Reply:
Yep! Completely agreed. I'd recommend suggesting both approaches and making explicit the consequences of each.

Reply:
I put a lot of thought into structuring and ordering. I hope ya'll like it 😄